docs/custom-stack.asciidoc

```diff
@@ -0,0 +1,262 @@
+==== Middleware error handler
+
+If you use the https://www.npmjs.com/package/connect[connect] module and an error is either thrown synchronously inside one of the middleware functions or is passed as the first argument to the middleware `next()` function,
+it will be passed to the https://www.npmjs.com/package/connect#error-middleware[Connect error handler].
+
+It's recommended that you register the agent as a Connect error handler.
+In the case where you have multiple Connect error handlers,
+the agent error handler should be the first in the chain to ensure that it will receive the error correctly.
+
+[source,js]
+----
+var apm = require('elastic-apm').start()
+var conncet = require('connect')
+
+var app = connect()
+
+// Your regular middleware and router...
+app.use(...)
+app.use(...)
+app.use(...)
+
+// Add the Elastic APM middleware after your regular middleware
+app.use(apm.middleware.connect())
+
+// ...but before any other error handler
+app.use(function (err, req, res, next) {
+  // Custom error handling goes here
+})
+----
+
+[[custom-stack-filter-sensitive-information]]
+=== Filter sensitive information
+
+By default the Node.js agent will filter common sensitive information before sending errors and metrics to the Elastic APM server.
+
+It's possible for you to tweak these defaults or remove any information you don't want to send to Elastic APM:
+
+* By default the Node.js agent will not log the body of HTTP requests.
+To enable this,
+use the <<log-body,`logBody`>> config option
+* By default the Node.js agent will filter certain HTTP headers known to contain sensitive information.
+To disable this,
+use the <<filter-http-headers,`filterHttpHeaders`>> config option
+* To apply custom filters,
+use the <<apm-add-filter,`apm.addFilter()`>> function
+
+[[custom-stack-add-your-own-data]]
+=== Add your own data
+
+The Node.js agent will keep track of the active HTTP request and will link it to errors and recorded transaction metrics when they are sent to the Elastic APM server.
+This allows you to see details about which request resulted in a particular error or which requests cause a certain HTTP endpoint to be slow.
+
+But in many cases,
+information about the HTTP request itself isn't enough.
+To add even more metadata to errors and transactions,
+use one of the two functions below:
+
+* <<apm-set-user-context,`apm.setUserContext()`>> - Call this to enrich collected performance data and errors with information about the user/client
+* <<apm-set-custom-context,`apm.setCustomContext()`>> - Call this to enrich collected performance data and errors with any information that you think will help you debug performance issues and errors (this data is only stored, but not indexed in Elasticsearch)
+* <<apm-set-tag,`apm.setTag()`>> - Call this to enrich collected performance data and errors with simple key/value strings that you think will help you debug performance issues and errors (tags are indexed in Elasticsearch)
+
```

Comment:
I haven't found any information that I have to provide paramters and in which form they are expected for
`apm.setUserContext()`, `apm.setCustomContext()`, `apm.setTag()`

Reply:
If you follow the links for each of the functions you should should a good explanation for how to use them.

Reply:
I completely overlooked the links, thanks.